### PR TITLE
[GST-649] Makes htpasswd step optional

### PR DESCRIPTION
**Pull request by lobsterdore**
This is so the htpasswd file can be controlled by another role.
By specifying an empty set of users you can now skip generation
the passwd file completely.

### CODE REVIEW

**Comment by quater** [comment]
LGTM :+1:

File: tasks/configure.yml [hunk 1]
```diff
@@ -45,18 +45,23 @@
...
 
+- name: Ensures go-server user passwords file
+  become: yes
+  template:
+    dest: "/home/{{ gocd_server.user }}/passwd"
+    group: "{{ gocd_server.group }}"
+    mode: 0600
+    owner: "{{ gocd_server.user }}"
+    src: passwd.j2
+  when: "{{ gocd_server.passwd_users | length > 0 }}"
```
Comment (wojtek-oledzki) [inline tasks/configure.yml]: can we loose the `{{`. It should work with just `when: gocd_server.passwd_users | length > 0`

Comment (wojtek-oledzki) [reply tasks/configure.yml]: and a bit of README update would be much appreciated :)

Comment (lobsterdore) [reply tasks/configure.yml]: So there is a bit in the readme that states that you need to specify users to manage the file, so it kind of already fits in, do you want to add something specific though? And yes I will fix the brackets, not sure why I put them in!

**Pull request merged**